src/main/kotlin/io/sc3/goodies/misc/StairWrenchItem.kt

```diff
@@ -0,0 +1,88 @@
+package io.sc3.goodies.misc
+
+import io.sc3.goodies.util.BaseItem
+import io.sc3.text.color
+import net.minecraft.block.Block
+import net.minecraft.block.BlockState
+import net.minecraft.block.StairsBlock
+import net.minecraft.client.item.TooltipContext
+import net.minecraft.entity.player.PlayerEntity
+import net.minecraft.item.ItemStack
+import net.minecraft.item.ItemUsageContext
+import net.minecraft.server.network.ServerPlayerEntity
+import net.minecraft.state.property.Property
+import net.minecraft.text.Text
+import net.minecraft.text.Text.translatable
+import net.minecraft.util.ActionResult
+import net.minecraft.util.Formatting
+import net.minecraft.util.Hand
+import net.minecraft.util.Util
+import net.minecraft.util.math.BlockPos
+import net.minecraft.world.World
+import net.minecraft.world.WorldAccess
+
+class StairWrenchItem(settings: Settings) : BaseItem(settings) {
+  private var properties: List<Property<*>> = listOf(StairsBlock.FACING, StairsBlock.HALF, StairsBlock.SHAPE)
+
+  override fun canMine(state: BlockState?, world: World, pos: BlockPos?, miner: PlayerEntity): Boolean {
+    if (!world.isClient) {
+      this.use(miner, state ?: return false, world, pos ?: return false, false, miner.getStackInHand(Hand.MAIN_HAND))
```

Comment:
a quick look at the two places `canMine` is called shows that both `state` and `pos` are definitely non-null

```suggestion
  override fun canMine(state: BlockState, world: World, pos: BlockPos, miner: PlayerEntity): Boolean {
    if (!world.isClient) {
      this.use(miner, state, world, pos, false, miner.getStackInHand(Hand.MAIN_HAND))
```

Reply:
learning the lesson that the java to kotlin converter has no idea what it's doing (which is bad because neither do I)

Reply:
it's not that it doesn't know what it's doing, it's just that it's assuming that types are nullable when they aren't explicitly marked as nullable or non-null, and it can't be inferred. it always errs on the safe side, because otherwise it'll lead to runtime NPEs